apps/dashboard/src/components/contract-functions/interactive-abi-function.tsx

```diff
@@ -352,22 +424,32 @@ export const InteractiveAbiFunction: React.FC<InteractiveAbiFunctionProps> = ({
               rightIcon={<Icon as={FiPlay} />}
               colorScheme="primary"
               isLoading={readLoading}
-              type="submit"
+              onClick={handleContractRead}
               form={formId}
             >
               Run
             </Button>
           ) : (
-            <TransactionButton
-              isDisabled={!abiFunction}
-              colorScheme="primary"
-              transactionCount={1}
-              isLoading={mutationLoading}
-              type="submit"
-              form={formId}
-            >
-              Execute
-            </TransactionButton>
+            <>
+              <ToolTipLabel label="Simulate the transaction to see its potential outcome without actually sending it to the network. This action doesn't cost gas.">
+                <Button
+                  onClick={handleContractSilumation}
+                  isDisabled={!abiFunction}
+                >
+                  Simulate
```

Comment:
you need a loading state for this, also i think when it fails right now it just doesnt show anything